src/StructIO.jl

```diff
@@ -311,4 +325,50 @@ function pack(io::IO, source::T, endianness::Symbol = :NativeEndian) where {T}
     return nothing
 end
 
+function strip_gap_nodes!(strct_expr::Expr)
+    @assert strct_expr.head == :struct
+    strct_nodes = strct_expr.args[3].args
+    K = length(strct_nodes)
+    gap_locations = UInt[]
+    # accumulator, in case several gap specifications occur in unbroken succession
+    gap_value = zero(UInt)
+    # gaps before structure fields
+    field_gap_values = UInt[]
+    for k in 1:K
+        f = strct_nodes[k]
+        !isa(f, LineNumberNode) || continue
+        if f.head === :(::)
```

Comment:
Since you have nothing after this conditional, you could simplify the condition to check that `f` is an `Expr` and that it has `::` as its head, which will naturally skip `LineNumberNode`s.
```suggestion
        if Meta.isexpr(f, :(::))
```